src/components/macro/macro-popover/macro-popover.component.ts

```diff
@@ -0,0 +1,118 @@
+import {Component, OnInit, OnChanges, Input, Output, EventEmitter, ViewChild } from '@angular/core';
+import {NgIf, NgSwitch, NgSwitchCase } from '@angular/common';
+
+import {Tab} from '../../popover/tab/tab';
+
+import {KeypressTabComponent} from '../../popover/tab/keypress/keypress-tab.component';
+import {MouseTabComponent} from '../../popover/tab/mouse/mouse-tab.component';
+
+import {MacroTextTabComponent} from './tab/macro-text/macro-text.component';
+import {MacroDelayTabComponent} from './tab/macro-delay/macro-delay.component';
+import {MacroMouseTabComponent} from './tab/macro-mouse/macro-mouse.component';
+
+import {MacroItemComponent} from '../../popover/tab/macro/macro-item.component';
+import {MacroAction, macroActionType} from '../../../../config-serializer/config-items/MacroAction';
+
+enum TabName {
+    Keypress,
+    Text,
+    Mouse,
+    Delay
+};
+
+@Component({
+    moduleId: module.id,
+    selector: 'macro-popover',
+    template: require('./macro-popover.component.html'),
+    styles: [
+        require('../../popover/popover.component.scss'),
+        require('./macro-popover.component.scss')
+    ],
+    host: { 'class': 'popover macro-action' },
+    directives: [
```

Comment:
Don't add directives here. It is deprecated since rc.5 add them in app.module.ts